torch/nn/functional.py

```diff
@@ -699,6 +705,10 @@ def _max_pool1d(
     return torch.max_pool1d(input, kernel_size, stride, padding, dilation, ceil_mode)
 
 
+# Used to limit number of warnings in CI to only one
+# See: https://github.com/pytorch/pytorch/issues/71257 for more context
+max_pool1d_with_indices.warning_count = 0
```

Comment:
I think you don't want to store these on the function object. That's why jit is not happy.
Just make these proper globals: `_MAX_POOL1D_WITH_INDICES_WARN_COUNT = 0`

Reply:
Thanks, @albanD for the suggestion. I updated the PR with the changes, but the tests fail again. I'm not really sure about this. cc: @eellison 

Also, using `with warnings.catch_warnings()` doesn't solve the spam. It still creates 1000s of warnings for `python3 test/test_ops.py -k max_pool<N>d`.

Reply:
Well, looks like jit doesn't support global either...
You can always guard the warning with a `if not torch.jit.is_scripting():` to hide that code from jit. People running script won't see the warning, but I think that's fine.